### PR TITLE
feat(apple): create an updatable pass (#53)

### PR DESCRIPTION
**Pull request by aahna-ashina**
## Dework Task

https://app.dework.xyz/nation3/app-2?taskId=063a5647-479d-47aa-ba6a-fbf3369acd8e

## Related GitHub Issue

close https://github.com/nation3/mobile-passport/issues/53

### Features Implemented

- [x] https://github.com/nation3/mobile-passport/pull/55/commits/82bb53ff0f8b0f20ae45a62ac4596640d93e954f Add 2nd template version
- [x] https://github.com/nation3/mobile-passport/pull/55/commits/54f5299d54bb7256de371bed3443eb3f99bfcd56 Add a web service to update the pass
- [x] https://github.com/nation3/mobile-passport/pull/55/commits/545e4b4e5289b9efd5d3126fa430b7f8ecf73c56 Set the shared secret (authentication token) to be used with the web service

## How Has This Been Tested?

- [x] Test generation of new `.pkass` files, and verify that `pass.json` contains the two new variables.
- [x] Install generated pass on iOS device, and verify that a request to `webServiceURL` was made
  request

## Remaining Admin Tasks

Tasks to handle _before_ merging this PR:

- [x] Add `APPLE_AUTH_TOKEN_HMAC_SEED` environment variable to Vercel deployment for https://passports.nation3.org/
  **Note:**  _Different_ from the value in `.env.local.goerli`
- [x] Add `APPLE_WEB_SERVICE_URL` environment variable to Vercel deployment for https://passports.nation3.org/
  **Note:**  _Same_ as the value in `.env.local.mainnet`

### CODE REVIEW

**Comment by luisivan** [comment]
@aahna-ashina can you provide more info about how to obtain `APPLE_AUTH_TOKEN_HMAC_SEED ` and `APPLE_WEB_SERVICE_URL`? Thanks!

**Comment by aahna-ashina** [comment]
> can you provide more info about how to obtain `APPLE_AUTH_TOKEN_HMAC_SEED ` and `APPLE_WEB_SERVICE_URL`? Thanks!

@luisivan Yes 😄

- The `APPLE_AUTH_TOKEN_HMAC_SEED ` variable can be any random string of characters, it just needs to be _different_ from the one used in [`.env.local.sample`](https://github.com/nation3/mobile-passport/pull/55/files#diff-75c9561a6558a85cc1b0645ec55fc66282778fb8e44cbc62e21da827f9a865e5R9)
  - Also note that once set, this variable should never be changed.

- The `APPLE_WEB_SERVICE_URL` must match the domain used for the Vercel deployment:
  - `APPLE_WEB_SERVICE_URL=https://passports.nation3.org/api/apple`

**Comment by luisivan** [comment]
@aahna-ashina thanks, done!

File: server/pages/api/pushUpdate.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
   // Push update of new template
   console.log('Pushing template update...')
 
-  const templateFormatVersion: number = 1
+  const templateFormatVersion: number = 2
```
Comment (johnmark13) [inline server/pages/api/pushUpdate.ts:11]: I didn't think much of the 2 in downloadPass.ts, but assuming it relates to this 2 here, should probably be in config somewhere.

**Review by johnmark13 — CHANGES_REQUESTED** [review]
Tiny change, always great work 💪

File: server/pages/api/pushUpdate.ts (hunk 1 above)
Comment (aahna-ashina) [reply server/pages/api/pushUpdate.ts:11]: @johnmark13 Yes, absolutely agreed.  Will handle that when I get started on #6

**Review by johnmark13 — APPROVED** [review]

**Review by luisivan — APPROVED** [review]
Solid

**Pull request merged**